File: app/src/main/java/org/mifos/mobilewallet/auth/domain/model/NewUser.java
package org.mifos.mobilewallet.auth.domain.model;

/**
 * Created by naman on 17/6/17.
 */

public class NewUser {

    private String username;
    private String email;
    private String firstname;
    private String lastname;
    private String officeId;
    private String roles;
    private String sendPasswordToEmail;

    public String getEmail() {
        return email;
    }

    public String getFirstname() {
        return firstname;
    }

    public String getLastname() {
        return lastname;
    }

    public String getUsername() {
        return username;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    public void setUsername(String username) {
        this.username = username;
    }
}

File: core/model/src/main/java/com/mifos/mobilewallet/model/domain/user/NewUser.kt
package com.mifos.mobilewallet.model.domain.user

import java.util.Collections

class NewUser(
    val username: String?,
    val firstname: String?,
    val lastname: String?,
    val email: String?,
    val password: String?
) {
    val officeId = "1"
    val roles: MutableList<Int> = NEW_USER_ROLE_IDS.toMutableList()
    val sendPasswordToEmail = false
    val isSelfServiceUser = true
    val repeatPassword: String? = password
}

private const val MOBILE_WALLET_ROLE_ID = 471
private const val SUPER_USER_ROLE_ID = 1

val NEW_USER_ROLE_IDS: Collection<Int> = Collections.unmodifiableList(
    listOf(MOBILE_WALLET_ROLE_ID, SUPER_USER_ROLE_ID)
)
